openedx/core/djangoapps/coursegraph/management/commands/dump_to_neo4j.py

```diff
@@ -166,59 +167,80 @@ def add_to_transaction(neo4j_entities, transaction):
         for entity in neo4j_entities:
             transaction.create(entity)
 
-    def handle(self, *args, **options):  # pylint: disable=unused-argument
+    @staticmethod
+    def should_dump_course(course_key):
         """
-        Iterates through each course, serializes them into graphs, and saves
-        those graphs to neo4j.
+        Only dump the course if it's been changed since the last time it's been
+        dumped.
+        :param course_key: a CourseKey object.
+        :return: bool. Whether or not this course should be dumped to neo4j.
         """
-        host = options['host']
-        port = options['port']
-        neo4j_user = options['user']
-        neo4j_password = options['password']
 
-        authenticate(
-            "{host}:{port}".format(host=host, port=port),
-            neo4j_user,
-            neo4j_password,
+        last_this_command_was_run = COMMAND_LAST_RUN_CACHE.get(course_key)
```

Comment:
That's interesting... I think it would be possible to store when the command was last run in coursegraph, but will be trickier to store when the course was last published. We don't currently store neo4j credentials, so we're not able to write to it outside of this management command, which means we wouldn't be able to write to it in a course published signal handler.

It might generally be useful to know when the modulestore was last dumped to neo4j, but let's put that in the next version. For now, I think it's ok to keep using caches, unless you have a specific reason we shouldn't use them.